python/ray/train/_internal/backend_executor.py

```diff
@@ -593,18 +574,19 @@ def get_with_failure_handling(self, remote_values):
         Returns:
             The resolved objects represented by the passed in ObjectRefs.
         """
-        success, exception = check_for_failure(remote_values)
-        if success:
-            return ray.get(remote_values)
-        else:
-            self._last_failure = exception
-            self._increment_failures()
-            logger.warning(
-                "Failure identified during training. Restarting all workers and "
-                "continuing training from latest checkpoint."
-            )
-            self._restart()
```

Comment:
That's because the specially handled `RayActorError` in this else case would raise during `_increment_failures` anyways, since `max_retries=0`